src/Stryker.Core/Stryker.Core/Mutators/IntegerNullificationMutator.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Stryker.Core.Mutants;
+
+namespace Stryker.Core.Mutators;
+
+public class IntegerNullificationMutator : MutatorBase<LiteralExpressionSyntax>
+{
+    public override MutationLevel MutationLevel => MutationLevel.Standard;
+
+    public override IEnumerable<Mutation> ApplyMutations(LiteralExpressionSyntax node, SemanticModel semanticModel)
+    {
+        if (IsNumberLiteral(node) && node.Token.Value is int value && value != 0)
+        {
+            yield return new Mutation
+            {
+                OriginalNode = node,
+                ReplacementNode = SyntaxFactory.LiteralExpression(SyntaxKind.NumericLiteralExpression, SyntaxFactory.Literal(0)),
+                DisplayName = "Integer nullification mutation",
+                Type = Mutator.Number
+            };
+        }
+    }
+
+    private static bool IsNumberLiteral(LiteralExpressionSyntax node)
+    {
+        var kind = node.Kind();
+        return kind == SyntaxKind.NumericLiteralExpression && node.Parent is EqualsValueClauseSyntax;
```

Comment:
why limiting this mutation to assignment ? it seems that it could be uses within any expression. It should at least also support method arguments.

Reply:
great idea! initially, I didn't have a constraint on the node parent at all so all occurrences would be mutated. I then constrained it to at least work for the example in the original issue, and I kinda left it open for suggestions if other places should receive this mutation as well

Reply:
limitation should only be placed if one detect an actual issue. But most (if not all) risky constructs, such as constants or attributes, are already identified and dealt with.